Web Inspector: add test harness for ECMA426 spec tests

takikawa opened the pull request:
#### cad38b25f854fedb50a2ecd5b884ffd10e9647f5
<pre>
Web Inspector: add test harness for ECMA426 spec tests

<a href="https://bugs.webkit.org/show_bug.cgi?id=284680">https://bugs.webkit.org/show_bug.cgi?id=284680</a>

Reviewed by Devin Rousso.

This patch adds a layout test for the web inspector using the imported spec
tests from the test repo for the current Source Map specification (ECMA426):

  <a href="https://github.com/tc39/ecma426">https://github.com/tc39/ecma426</a>
  <a href="https://github.com/tc39/source-map-tests">https://github.com/tc39/source-map-tests</a>

The test cases themselves can be found as imported tests at
`LayoutTests/imported/tc39-tg4`.

Also adds an update script that can import new versions of the tests
from a git repo &amp; branch (by default the upstream one).

As the test expectations file shows, most of the tests pass as well. There are
only a handful of cases where the spec tests and web inspector don&apos;t quite
agree yet:
  - sourcesContentNotStringOrNull: this is due to a test typo, to be fixed upstream
  - sourceRoot resolution tests: due to differences in handling URLs and sourceRoot
  - mappingSemanticsSingleFieldSegment: for a 1-segment mapping

* LayoutTests/imported/tc39-tg4/source-map-tests/LICENSE.md: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/README.WebKit: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/README.md: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/range-mappings-proposal-tests.json: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/basic-mapping-as-index-map.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/basic-mapping-as-index-map.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/basic-mapping-original.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/basic-mapping.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/basic-mapping.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/file-not-a-string-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/file-not-a-string-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/file-not-a-string-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/file-not-a-string-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-empty.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-empty.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-out-of-bounds-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-out-of-bounds-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-out-of-bounds-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-out-of-bounds-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-valid-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-valid-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-3.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-3.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-4.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-4.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-empty-sections.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-empty-sections.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-file-wrong-type-1.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-file-wrong-type-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-file-wrong-type-2.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-file-wrong-type-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-base-mappings.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-base-mappings.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-order.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-order.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-overlap.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-overlap.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-sub-map.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-sub-map.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-file.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-file.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-map.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-map.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-offset-column.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-offset-column.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-offset-line.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-offset-line.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-offset.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-offset.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-offset-column-wrong-type.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-offset-column-wrong-type.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-offset-line-wrong-type.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-offset-line-wrong-type.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-two-concatenated-sources.js: Added.
(foo):
(bar):
(baz):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-two-concatenated-sources.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-wrong-type-map.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-wrong-type-map.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-wrong-type-offset.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-wrong-type-offset.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-wrong-type-sections.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-wrong-type-sections.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-bad-separator.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-bad-separator.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-not-a-string-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-not-a-string-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-not-a-string-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-not-a-string-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-column-too-large.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-column-too-large.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-name-index-out-of-bounds.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-name-index-out-of-bounds.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-name-index-too-large.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-name-index-too-large.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-column.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-column.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-name-index.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-name-index.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-original-column.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-original-column.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-original-line.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-original-line.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-column.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-column.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-name-index.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-name-index.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-original-column.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-original-column.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-original-line.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-original-line.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-source-index.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-source-index.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-source-index.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-source-index.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-original-column-too-large.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-original-column-too-large.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-original-line-too-large.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-original-line-too-large.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-source-index-out-of-bounds.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-source-index-out-of-bounds.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-source-index-too-large.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-source-index-too-large.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-with-three-fields.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-with-three-fields.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-with-two-fields.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-with-two-fields.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-with-zero-fields.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-with-zero-fields.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-vlq-missing-continuation.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-vlq-missing-continuation.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-vlq-non-base64-char-padding.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-vlq-non-base64-char-padding.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-vlq-non-base64-char.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-vlq-non-base64-char.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-column-reset.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-column-reset.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-five-field-segment.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-five-field-segment.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-four-field-segment.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-four-field-segment.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-relative-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-relative-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-relative-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-relative-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-single-field-segment.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-single-field-segment.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mappings-missing.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mappings-missing.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-missing.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-missing.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-not-a-list-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-not-a-list-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-not-a-list-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-not-a-list-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-not-string.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-not-string.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/empty-string.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/empty-string.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/invalid-base64-char-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/invalid-base64-char-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/invalid-base64-char-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/invalid-base64-char-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/newline-semantics.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/newline-semantics.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/non-full-line-coverage.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/non-full-line-coverage.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/out-of-range.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/out-of-range.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/simple.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/simple.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/wrong-type-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/wrong-type-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/wrong-type-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/wrong-type-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/wrong-type-3.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/wrong-type-3.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/second-source-original.js: Added.
(baz):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-resolution-absolute-url.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-resolution-absolute-url.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-root-not-a-string-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-root-not-a-string-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-root-not-a-string-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-root-not-a-string-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-root-resolution.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-root-resolution.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-and-sources-content-both-null.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-and-sources-content-both-null.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-missing.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-missing.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-not-a-list-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-not-a-list-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-not-a-list-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-not-a-list-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-not-string-or-null.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-not-string-or-null.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-missing.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-missing.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-non-null-sources-content-null.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-non-null-sources-content-null.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-not-a-list-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-not-a-list-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-not-a-list-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-not-a-list-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-not-string-or-null.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-not-string-or-null.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-null-sources-content-non-null.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-null-sources-content-non-null.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/transitive-mapping-original.js: Added.
(foo):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/transitive-mapping-original.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/transitive-mapping-three-steps.js: Added.
(foo):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/transitive-mapping-three-steps.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/transitive-mapping.js: Added.
(foo):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/transitive-mapping.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/typescript-original.ts: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/unrecognized-property.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/unrecognized-property.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-boundary-values.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-boundary-values.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-empty-groups.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-empty-groups.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-empty-string.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-empty-string.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-large-vlq.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-large-vlq.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-missing.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-missing.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-not-a-number.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-not-a-number.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-numeric-string.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-numeric-string.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-too-high.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-too-high.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-too-low.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-too-low.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-valid.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-valid.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-continuation-bit-present-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-continuation-bit-present-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-continuation-bit-present-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-continuation-bit-present-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-negative-digit.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-negative-digit.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-single-digit.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-single-digit.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/source-map-spec-tests.json: Added.
* LayoutTests/inspector/model/source-map-spec-expected.txt: Added.
* LayoutTests/inspector/model/source-map-spec.html: Added.
* Tools/Scripts/import-source-map-tests: Added.
(generate_readme_contents):

Canonical link: <a href="https://commits.webkit.org/294514@main">https://commits.webkit.org/294514@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7c555c3c4990fbd22da0bc6830f425a6e39fd7a0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101864 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21532 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11848 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/107022 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/52498 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/103904 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21840 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/30039 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77567 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34574 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104871 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/16875 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91980 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57893 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16701 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/10003 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/51850 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86553 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10078 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109381 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/28997 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21357 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86538 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29358 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88184 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86114 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21949 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30865 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8587 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/23168 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28925 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/28736 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/32059 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30295 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on LayoutTests/imported/tc39-tg4/source-map-tests/README.md:
```diff
@@ -0,0 +1,55 @@
+The tests for Firefox are in the Mozilla [source-map](https://github.com/mozilla/source-map) library:
+  * The upstream repo has a [test file](https://github.com/mozilla/source-map/blob/master/test/test-spec-tests.js) for running the spec tests from this repo. They can be run with `npm test`.
+
+How to run in WebKit:
+  * Check out [WebKit](https://github.com/WebKit/WebKit/)
+  * `cd` to the checked out WebKit directory.
+  * Run `git am <this-repo>/webkit/0001-Add-harness-for-source-maps-spec-tests.patch`
+  * Run `Tools/Scripts/build-webkit` (depending on the platform you may need to pass `--gtk` or other flags)
+  * Run `Tools/Scripts/run-webkit-tests LayoutTests/inspector/model/source-map-spec.html` (again, you may need `--gtk` on Linux)
```
should this be updated now or later to be upstreamed here?

takikawa replied:
I'm planning to update this upstream once this patch makes it in. Since there's a couple other test tweaks I plan to make, I can PR all of those together in a test update patch.

dcrousso commented on LayoutTests/inspector/model/source-map-spec.html:
```diff
@@ -0,0 +1,112 @@
+    let suite = InspectorTest.createAsyncSuite("SourceMapSpec");
+
+    suite.addTestCase({
+        name: "SourceMapSpec",
```
on average how long does this test take to run?

all Web Inspector test cases have a default timeout of 10s, so will this run up against that or is it quicker?

have you tried running it in debug and/or using WK1 (i.e. legacy WebKit)?

takikawa replied:
> on average how long does this test take to run?

It seems to take about 3.4s on release and 6.5s on debug builds according to `--details`, at least on my M2 mac.

But I think I've now made the timeout issue moot, because after trying a bunch of different approaches I was able to finally split the big mono test case into multiple test cases. I'll comment below showing where the code changed for this. IIUC this should avoid the timeout problem since individual test cases each have their own timeout.

dcrousso commented on LayoutTests/imported/tc39-tg4/source-map-tests/LICENSE.md:
```diff
@@ -0,0 +1,14 @@
+The Source Map Tests suite ("Software") is protected by copyright and is being made available under the  "BSD License", included below. This Software may be subject to third party rights (rights from parties other than Ecma International), including patent rights, and no licenses under such third party rights are granted under this license even if the third party concerned is a member of Ecma Int ...
+
+Copyright (c) 2024, Ecma International
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING ...
+
+* Ecma International Standards hereafter means Ecma International Standards as well as Ecma Technical Reports
```
@jdatapple do you know if this license is problematic for WebKit?  it's almost identical to the license already used by [`JSTests/test262/LICENSE`](https://github.com/WebKit/WebKit/blob/main/JSTests/test262/LICENSE) so hopefully it's fine?

kmiller68 replied:
As I understand it, BSD is fine for WebKit. Most code is BSD and these are tests, not distributed code anyway.

dcrousso reviewed (commented):
generally this all looks fine to me (tho i dunno specifically how we normally write python scripts)

assuming the license is fine and there's no issues with how long it takes to run, im supportive :)

takikawa commented on LayoutTests/inspector/model/source-map-spec.html:
```diff
@@ -0,0 +1,119 @@
+    document.body.appendChild(script);
+}
+
+function triggerDownloadingTestDescriptions() {
```
In the latest version of the patch, I used some AJAX here to fetch the test data. The test cases are generated and run in a callback for the GET, so it works fine without needing async functions or promises. This lets each source map test be its own web inspector TestSuite test case.

dcrousso reviewed (approved):
overall this looks awesome!  assuming there are no issues with the license, LGTM :)

once this is done i can take a look at addressing the few failures you found

gsnedders commented on Tools/Scripts/import-source-map-tests:
```diff
@@ -0,0 +1,106 @@
+
+from webkitpy.common.host import Host
+from webkitpy.common.system.filesystem import FileSystem
+from webkitpy.common.webkit_finder import WebKitFinder
```
I'd suspect this is all overkill here, and that you should just use the Python standard library instead, simply hardcoding the path from here to the LayoutTests.

gsnedders commented on Tools/Scripts/import-source-map-tests:
```diff
@@ -0,0 +1,106 @@
+from webkitpy.common.host import Host
+from webkitpy.common.system.filesystem import FileSystem
+from webkitpy.common.webkit_finder import WebKitFinder
+from webkitpy.common.checkout.scm.git import Git
```
Please don't use this; we really want to deprecate and remove this. You probably want to use `webkitscmpy` in preference to this.

gsnedders commented on Tools/Scripts/import-source-map-tests:
```diff
@@ -0,0 +1,106 @@
+    tests_path = host.filesystem.join(layout_tests_directory, "imported", "tc39-tg4", "source-map-tests")
+    git = Git(tests_path, None, executive=host.executive, filesystem=host.filesystem)
+    status_output = git.run([git.executable_name, "status", tests_path]).splitlines()
+    test_dir_clean = re.search("working tree clean", status_output[-1] if status_output else "")
```
Basically anything parsing the output of a git command is an anti-pattern; it would be better to do `git status --porcelain=v2 -z` and then check that the output is empty.

gsnedders commented on Tools/Scripts/import-source-map-tests:
```diff
@@ -0,0 +1,106 @@
+        checkout_folder = "source-map-tests"
+        checkout_directory = host.filesystem.join(temporary_directory, checkout_folder)
+        print("Checking out tests from " + source_map_repo_url + " into " + checkout_directory)
+        Git.clone(source_map_repo_url, checkout_directory, host.executive)
```
We probably want to make it possible for people to import from a local directory without cloning, too.

gsnedders reviewed (commented):
A few kinda small comments on the import script; but this otherwise looks great to have!

takikawa committed:
Web Inspector: add test harness for ECMA426 spec tests

https://bugs.webkit.org/show_bug.cgi?id=284680

Reviewed by Devin Rousso.

This patch adds a layout test for the web inspector using the imported spec
tests from the test repo for the current Source Map specification (ECMA426):

  https://github.com/tc39/ecma426
  https://github.com/tc39/source-map-tests

The test cases themselves can be found as imported tests at
`LayoutTests/imported/tc39-tg4`.

Also adds an update script that can import new versions of the tests
from a git repo & branch (by default the upstream one).

As the test expectations file shows, most of the tests pass as well. There are
only a handful of cases where the spec tests and web inspector don't quite
agree yet:
  - sourcesContentNotStringOrNull: this is due to a test typo, to be fixed upstream
  - sourceRoot resolution tests: due to differences in handling URLs and sourceRoot
  - mappingSemanticsSingleFieldSegment: for a 1-segment mapping

* LayoutTests/imported/tc39-tg4/source-map-tests/LICENSE.md: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/README.WebKit: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/README.md: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/range-mappings-proposal-tests.json: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/basic-mapping-as-index-map.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/basic-mapping-as-index-map.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/basic-mapping-original.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/basic-mapping.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/basic-mapping.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/file-not-a-string-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/file-not-a-string-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/file-not-a-string-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/file-not-a-string-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-empty.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-empty.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-out-of-bounds-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-out-of-bounds-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-out-of-bounds-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-out-of-bounds-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-valid-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-valid-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-3.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-3.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-4.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/ignore-list-wrong-type-4.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-empty-sections.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-empty-sections.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-file-wrong-type-1.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-file-wrong-type-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-file-wrong-type-2.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-file-wrong-type-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-base-mappings.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-base-mappings.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-order.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-order.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-overlap.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-overlap.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-sub-map.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-invalid-sub-map.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-file.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-file.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-map.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-map.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-offset-column.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-offset-column.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-offset-line.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-offset-line.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-offset.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-missing-offset.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-offset-column-wrong-type.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-offset-column-wrong-type.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-offset-line-wrong-type.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-offset-line-wrong-type.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-two-concatenated-sources.js: Added.
(foo):
(bar):
(baz):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-two-concatenated-sources.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-wrong-type-map.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-wrong-type-map.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-wrong-type-offset.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-wrong-type-offset.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-wrong-type-sections.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/index-map-wrong-type-sections.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-bad-separator.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-bad-separator.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-not-a-string-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-not-a-string-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-not-a-string-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-not-a-string-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-column-too-large.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-column-too-large.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-name-index-out-of-bounds.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-name-index-out-of-bounds.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-name-index-too-large.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-name-index-too-large.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-column.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-column.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-name-index.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-name-index.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-original-column.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-original-column.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-original-line.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-original-line.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-column.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-column.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-name-index.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-name-index.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-original-column.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-original-column.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-original-line.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-original-line.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-source-index.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-relative-source-index.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-source-index.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-negative-source-index.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-original-column-too-large.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-original-column-too-large.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-original-line-too-large.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-original-line-too-large.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-source-index-out-of-bounds.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-source-index-out-of-bounds.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-source-index-too-large.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-source-index-too-large.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-with-three-fields.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-with-three-fields.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-with-two-fields.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-with-two-fields.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-with-zero-fields.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-mapping-segment-with-zero-fields.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-vlq-missing-continuation.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-vlq-missing-continuation.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-vlq-non-base64-char-padding.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-vlq-non-base64-char-padding.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-vlq-non-base64-char.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/invalid-vlq-non-base64-char.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-column-reset.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-column-reset.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-five-field-segment.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-five-field-segment.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-four-field-segment.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-four-field-segment.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-relative-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-relative-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-relative-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-relative-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-single-field-segment.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mapping-semantics-single-field-segment.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mappings-missing.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/mappings-missing.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-missing.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-missing.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-not-a-list-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-not-a-list-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-not-a-list-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-not-a-list-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-not-string.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/names-not-string.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/empty-string.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/empty-string.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/invalid-base64-char-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/invalid-base64-char-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/invalid-base64-char-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/invalid-base64-char-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/newline-semantics.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/newline-semantics.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/non-full-line-coverage.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/non-full-line-coverage.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/out-of-range.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/out-of-range.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/simple.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/simple.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/wrong-type-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/wrong-type-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/wrong-type-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/wrong-type-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/wrong-type-3.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/proposals/range-mappings/wrong-type-3.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/second-source-original.js: Added.
(baz):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-resolution-absolute-url.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-resolution-absolute-url.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-root-not-a-string-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-root-not-a-string-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-root-not-a-string-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-root-not-a-string-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-root-resolution.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/source-root-resolution.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-and-sources-content-both-null.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-and-sources-content-both-null.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-missing.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-missing.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-not-a-list-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-not-a-list-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-not-a-list-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-not-a-list-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-not-string-or-null.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-content-not-string-or-null.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-missing.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-missing.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-non-null-sources-content-null.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-non-null-sources-content-null.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-not-a-list-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-not-a-list-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-not-a-list-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-not-a-list-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-not-string-or-null.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-not-string-or-null.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-null-sources-content-non-null.js: Added.
(foo):
(bar):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/sources-null-sources-content-non-null.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/transitive-mapping-original.js: Added.
(foo):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/transitive-mapping-original.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/transitive-mapping-three-steps.js: Added.
(foo):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/transitive-mapping-three-steps.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/transitive-mapping.js: Added.
(foo):
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/transitive-mapping.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/typescript-original.ts: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/unrecognized-property.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/unrecognized-property.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-boundary-values.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-boundary-values.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-empty-groups.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-empty-groups.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-empty-string.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-empty-string.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-large-vlq.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/valid-mapping-large-vlq.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-missing.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-missing.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-not-a-number.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-not-a-number.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-numeric-string.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-numeric-string.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-too-high.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-too-high.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-too-low.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-too-low.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-valid.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/version-valid.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-continuation-bit-present-1.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-continuation-bit-present-1.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-continuation-bit-present-2.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-continuation-bit-present-2.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-negative-digit.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-negative-digit.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-single-digit.js: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/resources/vlq-valid-single-digit.js.map: Added.
* LayoutTests/imported/tc39-tg4/source-map-tests/source-map-spec-tests.json: Added.
* LayoutTests/inspector/model/source-map-spec-expected.txt: Added.
* LayoutTests/inspector/model/source-map-spec.html: Added.
* Tools/Scripts/import-source-map-tests: Added.
(generate_readme_contents):

Canonical link: https://commits.webkit.org/294514@main
